packages/@aws-cdk/aws-eks/lib/cluster.ts

```diff
@@ -769,3 +769,78 @@ class ImportedCluster extends Resource implements ICluster {
     }
   }
 }
+
+/**
+ * Properties for EksOptimizedAmi
+ */
+export interface EksOptimizedAmiProps {
+  /**
+   * What instance type to retrieve the image for (normal or GPU-optimized)
+   *
+   * @default Normal
+   */
+  readonly nodeType?: NodeType;
+
+  /**
+   * The Kubernetes version to use
+   *
+   * @default The latest version
+   */
+  readonly kubernetesVersion?: string;
+}
+
+/**
+ * Construct an Amazon Linux 2 image from the latest EKS Optimized AMI published in SSM
+ */
+export class EksOptimizedImage implements ec2.IMachineImage {
+  private readonly nodeType?: NodeType;
+  private readonly kubernetesVersion?: string;
+
+  private readonly amiParameterName: string;
+
+  /**
+   * Constructs a new instance of the EcsOptimizedAmi class.
+   */
+  public constructor(props: EksOptimizedAmiProps) {
+    this.nodeType = props && props.nodeType;
+    this.kubernetesVersion = props && props.kubernetesVersion || LATEST_KUBERNETES_VERSION;
+
+    // set the SSM parameter name
+    this.amiParameterName = `/aws/service/eks/optimized-ami/${this.kubernetesVersion}/`
+      + ( this.nodeType === NodeType.NORMAL ? "amazon-linux-2/" : "" )
+      + ( this.nodeType === NodeType.GPU ? " amazon-linux2-gpu/" : "" )
+      + "recommended/image_id";
+  }
+
+  /**
+   * Return the correct image
+   */
+  public getImage(scope: Construct): ec2.MachineImageConfig {
```

Comment:
Rename to “bind”

Reply:
Do you really want to refactor the `IMachineImage` interface?

https://github.com/aws/aws-cdk/blob/b0267e4b2dd61fa2e715cb5ab1787ef66cac0e2b/packages/%40aws-cdk/aws-ec2/lib/machine-image.ts#L9-L14

Reply:
oh, sorry. Didn't realize that's coming from there. Nevermind :-)